modules/ITK/cli/statismo-build-deformation-model.cxx

```diff
@@ -0,0 +1,142 @@
+#include <itkDirectory.h>
+#include <itkImageFileReader.h>
+
+#include "itkDataManager.h"
+#include "itkPCAModelBuilder.h"
+#include "itkStandardImageRepresenter.h"
+#include "itkStatisticalModel.h"
+
+#include <boost/program_options.hpp>
+
+#include "statismo-build-models-utils.h"
+
+namespace po = boost::program_options;
+
+
+struct programOptions {
+    bool bDisplayHelp;
+    string strDataListFile;
+    string strOutputFileName;
+    float fNoiseVariance;
+    bool bAutoNoise;
+};
+
+po::options_description initializeProgramOptions(programOptions& poParameters);
+bool isOptionsConflictPresent(programOptions& opt);
+void buildAndSaveDeformationModel(programOptions opt);
+
+
+
+int main(int argc, char** argv) {
+    programOptions poParameters;
+
+    po::positional_options_description optPositional;
+    optPositional.add("output-file", 1);
+    po::options_description optAllOptions = initializeProgramOptions(poParameters);
+
+
+    po::variables_map vm;
+    try {
+        po::parsed_options parsedOptions = po::command_line_parser(argc, argv).options(optAllOptions).positional(optPositional).run();
+        po::store(parsedOptions, vm);
+        po::notify(vm);
+    } catch (po::error& e) {
+        cerr << "An exception occurred while parsing the Command line:"<<endl;
+        cerr << e.what() << endl;
+        return EXIT_FAILURE;
+    }
+
+    if (poParameters.bDisplayHelp == true) {
+        cout << optAllOptions << endl;
+        return EXIT_SUCCESS;
+    }
+    if (isOptionsConflictPresent(poParameters) == true)	{
+        cerr << "A conflict in the options exists or insufficient options were set." << endl;
+        cout << optAllOptions << endl;
+        return EXIT_FAILURE;
+    }
+
+    try {
+        buildAndSaveDeformationModel(poParameters);
+    } catch (ifstream::failure & e) {
+        cerr << "Could not read the data-list:" << endl;
+        cerr << e.what() << endl;
+        return EXIT_FAILURE;
+    } catch (itk::ExceptionObject & e) {
+        cerr << "Could not build the model:" << endl;
+        cerr << e.what() << endl;
+        return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+}
+
+bool isOptionsConflictPresent(programOptions& opt) {
+    if (opt.strDataListFile == "" || opt.strOutputFileName == "" ) {
+        return true;
+    }
+
+    if (opt.strDataListFile == opt.strOutputFileName) {
+        return false;
+    }
+
+    return false;
+}
+
+void buildAndSaveDeformationModel(programOptions opt) {
+    const unsigned Dimensions = 3;
+    typedef itk::Image< itk::Vector<float, Dimensions>, Dimensions > ImageType;
+    typedef itk::StandardImageRepresenter<itk::Vector<float, 3>, 3> RepresenterType;
```

Comment:
@marcelluethi Could you please fact check me on this:
The first 3 (the vector dimensionality) is the number of color channels and the 2nd one is the image's dimensionality -> I'll replace them with 2 const unsigned to avoid magic numbers.